### PR TITLE
Add silence controller

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/14116

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: service/controller/resource/silence/create.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+					Name:    matcher.Name,
+					Value:   matcher.Value,
+				}
+
+				matchers = append(matchers, newMatcher)
+			}
+		}
+
+		newSilence := &alertmanager.Silence{
+			Comment:   silence.Name,
```
Comment (kopiczko) [inline service/controller/resource/silence/create.go:37]: I'd add here something silence-operator special. Like `"silence-operator: " + silence name`. To make sure there are no collisions. That should be propably encapsulated into some key function. Like `key.SilenceComment(silence)`.

This is important from my POV. Other comments not so much.

File: pkg/alertmanager/alertmanager.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	httpClient := &http.Client{}
+
+	return &AlertManager{
+		address:    config.Address,
+		httpClient: httpClient,
+	}, nil
+}
+
+func (am *AlertManager) GetSilenceByComment(comment string) (*Silence, error) {
+	silences, err := am.ListSilences()
```
Comment (kopiczko) [inline pkg/alertmanager/alertmanager.go:39]: Not sure if we should handle cases where there are mulitple silences with the same comment. Maybe for deletion.

File: pkg/alertmanager/alertmanager.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	}
+
+	if resp.StatusCode != 200 {
+		return microerror.Maskf(executionFailedError, fmt.Sprintf("failed to create silence %#q, expected code 200, got %d", s.Comment, resp.StatusCode))
+	}
+
+	return nil
+}
+
+func (am *AlertManager) DeleteSilenceByID(id string) error {
```
Comment (kopiczko) [inline pkg/alertmanager/alertmanager.go:73]: Feels like this object exposes more methods than necesarry. I think this can be private.

**Review by kopiczko — APPROVED** [review]

File: pkg/alertmanager/alertmanager.go (hunk 2 above)
Comment (corest) [reply pkg/alertmanager/alertmanager.go:39]: It is used during creation to check if the silence exists already. In future I think we can add silence ID into CR status after creation and then just get silence by ID from CR

**Pull request merged**